src/main/groovy/org/edx/jenkins/dsl/AnalyticsConstants.groovy

```diff
@@ -162,6 +162,23 @@ This text may reference other parameters in the task as shell variables, e.g.  $
         }
     }
 
+    public static def timeout_wrapper = { allVars ->
+        return {
+            timeout {
+                likelyStuck()
+                failBuild()
+            }
+        }
+    }
+
+    public static def terminate_cluster_post_build = { allVars ->
+        return {
+            postBuildTask {
+                task('Build timed out', 'cd $WORKSPACE/analytics-configuration && EXTRA_VARS="name=$CLUSTER_NAME" make terminate.emr', true)
```

Comment:
Oh, so the "Build timed out" is not a doc string, but a pattern that needs to match somewhere in the console log?   I didn't realize that.   I was going to suggest that maybe a comment here describing it would be helpful, but then noticed that there are no comments anywhere.

Reply:
Added a comment.